app/main.py

```diff
@@ -1,52 +1,27 @@
-import json
-import xml.etree.ElementTree as ET
+from app.book import Book
+from app.display import ConsoleDisplay, ReverseDisplay
+from app.printer import ConsolePrinter, ReversePrinter
+from app.serializer import JsonSerializer, XmlSerializer
 
-
-class Book:
-    def __init__(self, title: str, content: str):
-        self.title = title
-        self.content = content
-
-    def display(self, display_type: str) -> None:
-        if display_type == "console":
-            print(self.content)
-        elif display_type == "reverse":
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown display type: {display_type}")
-
-    def print_book(self, print_type: str) -> None:
-        if print_type == "console":
-            print(f"Printing the book: {self.title}...")
-            print(self.content)
-        elif print_type == "reverse":
-            print(f"Printing the book in reverse: {self.title}...")
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown print type: {print_type}")
-
-    def serialize(self, serialize_type: str) -> str:
-        if serialize_type == "json":
-            return json.dumps({"title": self.title, "content": self.content})
-        elif serialize_type == "xml":
-            root = ET.Element("book")
-            title = ET.SubElement(root, "title")
-            title.text = self.title
-            content = ET.SubElement(root, "content")
-            content.text = self.content
-            return ET.tostring(root, encoding="unicode")
-        else:
-            raise ValueError(f"Unknown serialize type: {serialize_type}")
+methods = {
+    "display": {"console": ConsoleDisplay, "reverse": ReverseDisplay},
+    "print": {"console": ConsolePrinter, "reverse": ReversePrinter},
+    "serialize": {"json": JsonSerializer, "xml": XmlSerializer}
+}
 
 
 def main(book: Book, commands: list[tuple[str, str]]) -> None | str:
+
     for cmd, method_type in commands:
         if cmd == "display":
-            book.display(method_type)
-        elif cmd == "print":
-            book.print_book(method_type)
-        elif cmd == "serialize":
-            return book.serialize(method_type)
+            display_action = methods.get("display").get(method_type)
+            display_action(book).display()
+        elif printer_action := methods.get("print").get(method_type):
+            printer_action(book).print_book()
+        elif serialize_action := methods.get("serialize").get(method_type):
+            return serialize_action(book).serialize()
```

Comment:
You don't need to use := because you don't use this variable and also why you don't use the same approach for cmd == display then?